pkg/cloudprovider/gcp/object_store.go

```diff
@@ -194,11 +215,37 @@ func (o *ObjectStore) DeleteObject(bucket, key string) error {
 	return errors.Wrapf(o.client.Bucket(bucket).Object(key).Delete(context.Background()), "error deleting object %s", key)
 }
 
+func (o *ObjectStore) SignBytes(bytes []byte) ([]byte, error) {
+	name := "projects/-/serviceAccounts/" + o.googleAccessID
+	payload := strings.Builder{}
+	if _, err := payload.Write(bytes); err != nil {
+		return nil, err
+	}
+	resp, err := o.iamSvc.Projects.ServiceAccounts.SignBlob(name, &iamcredentials.SignBlobRequest{
+		Delegates: []string{name},
+		Payload:   payload.String(),
+	}).Context(context.Background()).Do()
+
+	if err != nil {
+		return nil, err
+	}
+	data, err := base64.StdEncoding.DecodeString(resp.SignedBlob)
+	return data, err
+}
+
 func (o *ObjectStore) CreateSignedURL(bucket, key string, ttl time.Duration) (string, error) {
-	return storage.SignedURL(bucket, key, &storage.SignedURLOptions{
+	options := storage.SignedURLOptions{
 		GoogleAccessID: o.googleAccessID,
-		PrivateKey:     o.privateKey,
 		Method:         "GET",
 		Expires:        time.Now().Add(ttl),
-	})
+	}
+	if o.privateKey == nil {
+		options.SignBytes = func(b []byte) ([]byte, error) {
```

Comment:
I think you can just say `options.SignBytes = o.SignBytes` here

Reply:
It doesn't compile if I do it that way, since o.SignBytes is technically a method.

Reply:
`options.SignBytes` is a function pointer, so you should be able to assign a pointer to the `o.SignBytes` method to it via `options.SignBytes = o.SignBytes` -- but it can stay as-is.

Reply:
Yeah nvm, I was just dumb.